AdminWebsite/AdminWebsite/ClientApp/src/app/booking/hearing-schedule/hearing-schedule.component.ts

```diff
@@ -142,6 +144,24 @@ export class HearingScheduleComponent extends BookingBaseComponent implements On
       (this.hearingStartTimeHour.dirty || this.hearingStartTimeHour.touched || this.failedSubmission);
   }
 
+  startHoursInPast() {
+    const todayDate = new Date(new Date().setHours(0, 0, 0, 0));
+    const realDate = new Date(new Date(this.hearingDate.value).setHours(0, 0, 0, 0));
+    const todayHours = new Date().getHours();
+
+    this.isStartHoursInPast = realDate.toString() === todayDate.toString() && this.hearingStartTimeHour.value < todayHours
+      && (this.hearingStartTimeHour.dirty || this.hearingStartTimeHour.touched);
+  }
+
+  startMinutesInPast() {
+    const todayDate = new Date(new Date().setHours(0, 0, 0, 0));
+    const realDate = new Date(new Date(this.hearingDate.value).setHours(0, 0, 0, 0));
+    const todayHours = new Date().getHours();
+    const todayMinutes = new Date().getMinutes();
+    this.isStartMinutesInPast = realDate.toString() === todayDate.toString() && this.hearingStartTimeHour.value === todayHours
+      && this.hearingStartTimeMinute.value <= todayMinutes && (this.hearingStartTimeMinute.dirty || this.hearingStartTimeMinute.touched);
+  }
+
   get hearingStartTimeMinuteInvalid() {
```

Comment:
Might be a bit easier to do something like:

var now = new Date ();
var enteredDate = new Date ("2020-02-24");
var enteredHours = 12;
var enteredMinutes = 02;
enteredDate.setHours( enteredHours);
enteredDate.setMinutes( enteredMinutes);
alert ( enteredDate <= now );

Reply:
to validate time in the past, the start date should be today, otherwise we dont need to validate time for past . The entered date is dynamic (today), and could not be set manually.